samples/helpers/get-env/CMakeLists.txt

```diff
@@ -15,4 +15,6 @@ add_library(
       src/get_env.cpp
 )
 
+target_compile_definitions(get-env-helper PRIVATE _azure_BUILDING_SAMPLES)
```

Comment:
Why do we need to specify these `_azure_BUILDING_SAMPLES`, `_azure_BUILDING_SDK`, and `_azure_BUILDING_TESTS`?

This probably warrants a comment in one of the files (maybe in the azure-template CMakeLists.txt), to describe what we are doing. I can imagine most folks creating new files would miss adding this, so might be worth adding a doc section about these somewhere visible to contributors.